src/execution/mod.rs

```diff
@@ -161,7 +168,15 @@ pub(crate) fn run_main(
             timely::execute::execute_directly::<Result<(), String>, _>(move |worker| {
                 let (pump, probe) = Python::with_gil(|py| {
                     let flow = &flow.as_ref(py).borrow();
-                    build_dataflow(worker, py, flow, input_builder, output_builder)
+
+                    build_dataflow(
```

Comment:
The docs say it will just be slow. If it still works we could allow multiple processes locally (however slow) but a warning does seem prudent at minimum. If we're going to throw an error we should be more finger-waggy in the docs

Reply:
We could check the number of `addresses` (since each process gets one) and error out. It feels a little strange to prevent something that is correct but not performant. Where's your bar for "not performant"? Especially since any recovery store will have its performance trade offs and feels tough to encode all that into detectors and warnings.

Reply:
Mostly I figure if we say "only do this locally or it's kinda wrong" then we should also log a warning, but we haven't really formalized our log criteria so 🤷‍♀️